Tensile/Source/client/main.cpp

```diff
@@ -528,19 +530,18 @@ int main(int argc, const char* argv[])
                             auto kernels = solution->solve(problem, *inputs, *hardware);
 
                             size_t       warmupInvocations = listeners.numWarmupRuns();
```

Comment:
If ReferenceValidator::numWarmupRuns returns 2, we could run both the Front and Back modes in one process.

Reply:
@sdquiring Please check PR's third commit. 
I add a new mode "GuardPageAll". When config yaml set boundsCheck as "GuardPageAll", clientWriter would set numWarmups as 2. In new client main function, first run of warmups would go "GuardPageFront" checking and second run would go "GuardPageBack" checking.

Reply:
This new change breaks the encapsulation of the DataInitialization class.  It should be possible to implement this in the DataInitialization class.  If you need more info to be sent in (warmup number) then do that.